src/dev/flang/ast/Feature.java

```diff
@@ -1554,12 +1554,13 @@ private void checkChoiceAndAddInternalFields(Resolution res)
         p.calledFeature().checkNoClosureAccesses(res, p.pos());
       }
 
-    _choiceTag = new Feature(res,
-                             _pos,
+    _choiceTag = new Feature(_pos,
                              Visi.PRIVATE,
+                             Consts.MODIFIER_FIXED,
```

Comment:
The original problem here is that a tag field gets added to `choice` and to any feature that inherits from `choice`. Using `fixed` only masks the error without removing the duplicate field. 
My suggestion: change `FuzionConstants.CHOICE_TAG_NAME` to be just `"tag"` and add `priv int tag := ?` to `choice.fz`. 
If that does not work, alternatively, you could add the `_choiceTag` feature here, but only if `this` is `choice` (i..e, `this == Types.f_choice`, but I am not sure if `Types.f_choice` is set at this point).

Reply:
> My suggestion: change FuzionConstants.CHOICE_TAG_NAME to be just "tag" and add priv int tag := ? to choice.fz.
Unfortunately there are two checks that make this hard. First that choices do not contain fields, second that choices contain only declarations.